src/java.base/share/classes/sun/nio/ch/NativeSocketAddress.java

```diff
@@ -141,46 +136,22 @@ int encode(ProtocolFamily protocolFamily, InetSocketAddress isa) {
     InetSocketAddress decode() throws SocketException {
         int family = family();
         if (family != AF_INET && family != AF_INET6)
-            throw new SocketException("Socket family not recognized");
-        return new InetSocketAddress(address(family), port(family));
-    }
-
-    /**
-     * Find a mismatch between this and another socket address
-     * @return the byte offset of the first mismatch or -1 if no mismatch
-     */
-    private int mismatch(NativeSocketAddress other) {
-        int i = ArraysSupport.vectorizedMismatch(null,
-                this.address,
-                null,
-                other.address,
-                SIZEOF_SOCKETADDRESS,
-                ArraysSupport.LOG2_ARRAY_BYTE_INDEX_SCALE);
-        if (i >= 0)
-            return i;
-        i = SIZEOF_SOCKETADDRESS - ~i;
-        for (; i < SIZEOF_SOCKETADDRESS; i++) {
-            if (UNSAFE.getByte(this.address + i) != UNSAFE.getByte(other.address + i)) {
-                return i;
-            }
-        }
-        return -1;
+            throw new SocketException("Socket family not recognized: " + family);
+        var address = new InetSocketAddress(address(family), port(family));
+        return address;
     }
 
     @Override
     public boolean equals(Object other) {
-        if (other instanceof NativeSocketAddress) {
-            return mismatch((NativeSocketAddress) other) < 0;
-        } else {
-            return false;
-        }
+        return (other instanceof NativeSocketAddress otherNSA) &&
+                memory.mismatch(otherNSA.memory) < 0;
     }
 
     @Override
     public int hashCode() {
```

Comment:
This method can now delegate to `SegmentBulkOperations::contentHash,` a new method.